noggin/security/ipa.py

```diff
@@ -97,6 +98,35 @@ def otptoken_del(self, ipatokenuniqueid):
         data = self._request('otptoken_del', [], params)
         return data['result']
 
+    def otptoken_sync(self, user, password, first_code, second_code, token=None):
+        """
+        Sync an otptoken for a user.
+
+        :param user: the user to sync the token for
+        :type user: string
+        :param password: the user's password
+        :type password: string
+        :param first_code: the first OTP token
+        :type first_code: string
+        :param second_code: the second OTP token
+        :type second_code: string
+        :param token: the token description (optional)
+        :type token: string
+        """
+        data = {
+            'user': user,
+            'password': password,
+            'first_code': first_code,
+            'second_code': second_code,
+            'token': token,
+        }
+        try:
+            url = "https://" + self._host + "/ipa/session/sync_token"
+            response = requests.post(url=url, data=data, verify=self._verify_ssl)
+        except requests.RequestException:
+            raise
```

Comment:
I don't understand what this is for... If there is a `RequestException` you'll probably want to convert it to something else, no? Right now it does not do anything, the try..except is useless.

Reply:
So I wanted to just raise the exception so that the controller can handle it as I still have that logic there. Mainly because the controller has access to `app` so I can log from there. Is this a bad pattern? If I don't try...except here then it would make its way back to the controller anyway is it?

Reply:
Yes, uncaught exceptions will bubble up automatically. This pattern here is basically useless.
My thinking is that you may want to convert the RequestException into an exception that is more specific to `python_ipa`, probably `python_ipa.exceptions.BadRequest` or a more appropriate subclass of it. Then you would catch that exception in the controller.

Reply:
Makes sense. I've updated this file but I haven't transformed the error into a `python_freeipa` error as its not directly related to the client and its still a requests error. I don't think they will be moving from a http request for syncing.